provisioning/ubuntu-provision.sh

```diff
@@ -183,6 +183,15 @@ function install_docker() {
     "deb [arch=$(dpkg --print-architecture) signed-by=/usr/share/keyrings/docker-archive-keyring.gpg] https://download.docker.com/linux/ubuntu \
     $(lsb_release -cs) stable" | tee /etc/apt/sources.list.d/docker.list > /dev/null
   apt-get update --quiet
+
+  # Ensure that version is fixed
+  echo "
+Package: docker-ce*
+Pin: version 5:${DOCKER_VERSION}*
```

Comment:
What is the `5:` for?

Reply:
As far as I remember (to be searched and confirmed by the documentation), the package version prefix is the epoch. It is usually `0:` but in the case of `docker-ce` package it is `5: `. I assume it is some kind of trick to allow overriding at distribution level but I don't know really.